evmcore/tx_pool.go

```diff
@@ -1161,6 +1167,15 @@ func (pool *TxPool) requestPromoteExecutables(set *accountSet) chan struct{} {
 	}
 }
 
+// waitForIdleReorgLoop_forTesting allows tests to wait for the reorg loop to
+// finish its current run. This is useful for tests that want to control the
+// timing of reorgs and promotions, ensuring that the pool is in a stable state
+// before proceeding with further assertions or actions.
+func (pool *TxPool) waitForIdleReorgLoop_forTesting() {
```

Comment:
Nice comment. 
nit: considering this is a test only function, it should be in the corresponding testing file. Since it is in the same package it could take the txpool as a parameter and still refer to the channels.